imap/message.c

```diff
@@ -5731,6 +5731,10 @@ EXPORTED int message_get_encoding(message_t *m, int *encp)
 {
     int r = message_need(m, M_CACHEBODY);
     if (r) return r;
+
+    if (!m->body->charset_enc && m->body->encoding)
```

Comment:
This and the get_charset_id function changes suggest to me that we might want to get rid of `message_parse_cbodystructure` and replace it with calls to `message_read_bodystructure`. It's confusing to have two functions that load `struct body` from the cache but one of them only partially sets the message attributes.

I wonder if that is for optimization or only legacy from when the `message` API got implemented. Maybe @brong nows its backstory?

Depending on the urgency of this pull request, I could either approve this with a `TODO` added, or we give it a couple more time to try removing `message_parse_cbodystructure` if there is no good reason to keep it. I can help with that, if preferred.